apps/rush-lib/src/logic/pnpm/PnpmProjectDependencyManifest.ts

```diff
@@ -193,6 +194,53 @@ export class PnpmProjectDependencyManifest {
         // if it's been hoisted up as a top-level dependency
         const topLevelDependencySpecifier: DependencySpecifier | undefined =
           this._pnpmShrinkwrapFile.getTopLevelDependencyVersion(peerDependencyName);
+
+        if (!topLevelDependencySpecifier) {
+          // If it's an optional dependency, then it's okay if it's missing
+          if (shrinkwrapEntry.peerDependenciesMeta
+            && shrinkwrapEntry.peerDependenciesMeta.hasOwnProperty(peerDependencyName)) {
+            const peerDependenciesMetaYaml: IPeerDependenciesMetaYaml
+              = shrinkwrapEntry.peerDependenciesMeta[peerDependencyName];
+            if (peerDependenciesMetaYaml.optional) {
+              // We can safely ignore this missing dependency because it was marked as optional
+              continue;
+            }
+          }
+
+          // If it's one of the peer dependencies, then see if it got "hoisted"
+          // Parse "2.7.0_eslint@6.6.0+typescript@3.6.4" --> "eslint@6.6.0+typescript@3.6.4"
+          const split1: RegExpExecArray | null = /^[^_]+_([^_]+)$/.exec(specifier);
```

Comment:
This is a prototype.  If it works we need to extract this parsing logic into a proper function and make better variable names.